api/client_test.go

```diff
@@ -74,6 +74,29 @@ func TestSend(t *testing.T) {
 	}
 }
 
+func TestSendPost(t *testing.T) {
+	ts := mockStatus()
+	defer ts.Close()
+
+	type T struct {
+		ID string `json:"id"`
+	}
+
+	eg := T{ID: "id"}
+	in := T{}
+
+	err := api.NewClient(api.Endpoint(ts.URL)).
+		Post("/echo").Send(eg).Recv(&in)
```

Comment:
Keep in mind that `api.NewClient(api.Endpoint(ts.URL)).Post("/echo")` is a value. You can make a test

```
func TestSend(t *testing.T) {
   methods := []*CURL{
      api.NewClient(api.Endpoint(ts.URL)).Put("/echo"),
      api.NewClient(api.Endpoint(ts.URL)).Post("/echo"),
   }

  for _, method := range methods {
    eg := T{ID: "id"}
    in := T{}
    err := method.Send(eg).Recv(&in)

   // checks
  } 
}
```